app/controllers/api/v1/resources_controller.rb

```diff
@@ -19,25 +19,29 @@ def index
     filter_params = params[model.name].presence
 
     resources = Api::Index.new(model).where(filter_params).order(:id)
+    resources = resources.visible(api_key.user_id) if resources.respond_to? :visible
     resources = paginate_collection(resources)
     resources = decorate_collection(resources)
 
     render json: { model.name.pluralize => resources, :meta => resources.meta }
   end
 
   def show
-    resource = model.klass.find(params[:id])
-    render json: { model.name => decorate(resource) }
+    resource = model.klass.where(id: params[:id])
+    resource = resource.visible(api_key.user_id) if resource.respond_to? :visible
+    resource = resource.first
+    response_json = resource ? { model.name => decorate(resource) } : {}
```

Comment:
Shouldn't we return 403 or 401 instead of empty response? We do that in `destroy` action.